compiler/src/dotty/tools/dotc/transform/CheckLoopingImplicits.scala

```diff
@@ -84,7 +87,9 @@ class CheckLoopingImplicits extends MiniPhase:
         checkNotLooping(t.rhs)
       case _ =>
 
-    if sym.isOneOf(GivenOrImplicit | Lazy | ExtensionMethod) then
+    if sym.isOneOf(GivenOrImplicit | Lazy | ExtensionMethod)
+      || sym.name == nme.apply && sym.owner.is(Module) && sym.owner.sourceModule.isOneOf(GivenOrImplicit)
```

Comment:
Could you please add a comment to explain the intent behind the condition on the right side of the `||`? I wonder if this condition is too strong, but I am not sure. Do we have existing “pos” tests?